packages/playground/src/components/node_details.vue

```diff
@@ -0,0 +1,176 @@
+      >
+        <div style="display: flex; align-items: center">Failed to retrieve Node details.</div>
+        <template v-slot:append>
+          <v-icon @click="getNodeDetails" style="cursor: pointer">mdi-reload</v-icon>
+        </template>
+      </v-alert>
+    </div>
+  </td>
+
+  <td :colspan="columnsLen" v-else>
+    <v-row class="ma-2">
+      <v-col :cols="getColSize">
+        <div class="mt-3">
+          <card-details
+            :loading="false"
+            title="Node Resources"
+            icon="mdi-harddisk"
+            :items="NodeResourceFields"
+          ></card-details>
+        </div>
+      </v-col>
+
+      <v-col :cols="getColSize">
+        <div class="mt-3">
+          <card-details
+            :loading="false"
+            title="Location"
+            icon="mdi-map-marker"
+            :items="CountryResourceFields"
+          ></card-details>
+        </div>
+      </v-col>
+
+      <v-col :cols="getColSize" :class="{ 'mt-n8': getColSize() === 6 }">
+        <div class="mt-3">
+          <card-details
+            :loading="false"
+            title="Farm details"
+            icon="mdi-silo"
+            :items="FarmResourceFields"
+          ></card-details>
+        </div>
+      </v-col>
+    </v-row>
+    <v-row>
+      <v-col v-if="item.value.num_gpu > 0" cols="getColSize" style="min-width: 600px; min-height: 400px">
+        <div class="mt-3">
+          <GPUDetailsCard :node="node" />
+        </div> </v-col
+    ></v-row>
+  </td>
+</template>
+
+<script setup lang="ts">
+import type { GridNode } from "@threefold/gridproxy_client";
+import { onMounted, ref, watch } from "vue";
+
+import { getNode } from "@/explorer/utils/helpers";
+import type { NodeDetailsCard } from "@/explorer/utils/types";
+
+import { gridProxyClient } from "../clients";
+import { nodeInitializer } from "../explorer/utils/types";
+import toTeraOrGigaOrPeta from "../utils/toTeraOrGegaOrPeta";
+
+const dNodeError = ref(false);
+const dNodeLoading = ref(true);
+const farmName = ref("");
+const publicIps = ref(0);
+const NodeResourceFields = ref<NodeDetailsCard[]>();
+const CountryResourceFields = ref<NodeDetailsCard[]>();
+const FarmResourceFields = ref<NodeDetailsCard[]>();
+const node = ref<GridNode>(nodeInitializer);
+
+const props = defineProps({
+  item: {
+    required: true,
+    type: null,
+  },
+  columnsLen: {
+    required: true,
+    type: Number,
+  },
+});
+
+onMounted(async () => {
+  getNodeDetails();
+});
+
+async function getNodeDetails() {
+  try {
+    dNodeLoading.value = true;
+    const res = await gridProxyClient.farms.list({ farmId: props.item.value.farmId });
+    farmName.value = res.data[0].name;
+    publicIps.value = res.data[0].publicIps.length;
+    if (Array.isArray(res) && !res.length) throw new Error("Can't resolve farm data");
+    NodeResourceFields.value = getNodeResourceCard();
+    CountryResourceFields.value = getCountryResourceCard();
+    FarmResourceFields.value = getFarmResourceCard();
```

Comment:
If that's the case, you can call the function directly. Additionally, in TypeScript, the naming convention suggests using `camelCaseName`. For instance, instead of `NodeResourceFields`, it should be `nodeResourceFields`.